modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -930,7 +931,7 @@ public String getTypeDeclaration(Schema p) {
         Schema<?> target = ModelUtils.isGenerateAliasAsModel() ? p : schema;
         if (ModelUtils.isArraySchema(target)) {
             Schema<?> items = getSchemaItems((ArraySchema) schema);
-            return getSchemaType(target) + "<" + getTypeDeclaration(items) + ">";
+            return getSchemaType(target) + "<" + getBeanValidation(items) + getTypeDeclaration(items) + ">";
```

Comment:
Unfortunately, this is not sufficient when the type declaration is a fully qualified class name.

For example this schema

```json
"errors": {
  "type": "array",
  "items": { "$ref": "#/components/schemas/ProblemDetail" }
}
```

using this config

```xml
<schemaMappings>ProblemDetail=org.springframework.http.ProblemDetail</schemaMappings>
```

now generates invalid Java syntax:

```java
private List<@Valid org.springframework.http.ProblemDetail> errors;
```

According to the [JLS](https://docs.oracle.com/javase/specs/jls/se21/html/jls-4.html#jls-ClassType) this should be:

```java
private List<org.springframework.http.@Valid ProblemDetail> errors;
```